### PR TITLE
feat(runtime): log deprecated message on first usage of `window` global

### PR DESCRIPTION
**Pull request by dsherret**
Part of #13367

### CODE REVIEW

**Review by mmastrac — APPROVED** [review]
LGTM

**Review by lucacasonato — COMMENTED** [review]
Before doing this, we should add a `deno_lint` warning

**Comment by dsherret** [comment]
@lucacasonato there has been a `no-window-prefix` lint rule for quite some time: https://lint.deno.land/?q=no-window-prefix#no-window-prefix

**Comment by dsherret** [comment]
Oh, looks like we should make it more strict:

> ```ts
> // `alert` is allowed to call with `window` because it's not supported in Workers
> window.alert("🍣");
> ```

File: runtime/js/98_global_scope.js [hunk 1]
```diff
@@ -3,7 +3,9 @@
 const core = globalThis.Deno.core;
 const primordials = globalThis.__bootstrap.primordials;
 const {
+  ArraySplit,
   ObjectDefineProperties,
+  StringSlice,
```
Comment (petamoriken) [inline runtime/js/98_global_scope.js:8]: Probably `ArrayPrototypeSplit` and `StringPrototypeSlice` are collect.

**Comment by dsherret** [comment]
This is stale, so going to close.

**Pull request closed without merging**